### PR TITLE
CampFort - Check pokeballs before camping

### PR DESCRIPTION
**Pull request by davidakachaos**
## Short Description:
I noticed that the bot doesn't move when camping at a lure, but had run
out of pokeballs. This will first check IF we have any pokeballs left.
If we have no pokeballs left, abandon the lure (to move to a fort to get
more pokeballs) when we got some pokeballs again, will move back to the
lure.

### CODE REVIEW

File: pokemongo_bot/cell_workers/camp_fort.py [hunk 1]
```diff
@@ -29,6 +31,13 @@ def initialize(self):
         self.config_moving_time = self.config.get("moving_time", 600)
 
     def work(self):
+        # Let's make sure we have balls before we sit at a lure!
+        # See also catch_pokemon.py
+        if sum([inventory.items().get(ball.value).count for ball in
+            [Item.ITEM_POKE_BALL, Item.ITEM_GREAT_BALL, Item.ITEM_ULTRA_BALL]]) <= 0:
+            self.logger.info('Not any pokeballs left, refuse to sit at lure!')
+            return WorkerResult.ERROR
```
Comment (Anakin5) [inline pokemongo_bot/cell_workers/camp_fort.py]: That part should go after the check on "move_until". There is no point displaying that we would not sit there if the task is deactivated or if we are in the moving phase.
Also, does not seem like an error to me, more a WorkerResult.SUCCESS.

**Comment by Anakin5** [comment]
On top of my comment, I am wondering something. The CampFort is not staying at a specific fort but in the middle of a cluster, which makes it a good location to spin forts, lazily speaking.
I can image a scenario where you are always out of Pokeballs, and so the task would have no effect at all. 

The original goal was to really stop walking, take at rest at a convenient place. That we agree would be less efficient than walking around whole day, but also much more human.
Did you notice that you run out of Pokeballs while activating the CampFort ?

**Comment by davidakachaos** [comment]
I see what you mean. But my line of thought was; if I was outside, playing Pokemon Go, I'm sitting at a few lures, and I run out of Pokeballs... I would look around for other stops nearby (that might be the tweak here!) to spin and collect balls. As I would wander to other stops to get some balls, I would still come back to the lure "hotspot" when I got a few more balls.
What do you think? Is that a reasonable thing to argue here? Maybe with the distant tweak?

**Comment by Anakin5** [comment]
OK. Let's just move down a bit the code then, to avoid message spam.

**Comment by davidakachaos** [comment]
@Anakin5 Okay, so I've move the check below the enabled check. I also added a boolean to check if we've announced that the bot is out of balls. What do you think?

**Comment by Anakin5** [comment]
@davidakachaos Actually it should go down a little bit more, after the check `now < move_until`. That is because we don't want to warn about not staying at the fort if were going to do so in the first place (because the timer moving is still under way).

To limit the output, you could also activate the `move_until` timer otherwise we are going to get 1 ball at a nearby pokestop and then come back to camp with only 1 ball. Would be better to walk for 10 minutes.

**Comment by davidakachaos** [comment]
Thank you @Anakin5 for all the great advise :smile: Now the worker will move away from the lure for a set time after which it will check again for enough balls.

**Comment by solderzzc** [comment]
👍

**Comment by Anakin5** [comment]
I will do a little adjustement latter on to support move_time = 0 but it can wait a bit

**Pull request merged**